### PR TITLE
feat(common-codegen): create package

### PR DESCRIPTION
**Pull request by dk1a**
this package is meant for some utilities from `render-solidity`/`render-ts`
not sure what's happening with pnpm-lock.yaml, I just ran `pnpm install` with pnpm@8.2.0

### CODE REVIEW

File: packages/codegen-utils/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@latticexyz/codegen-utils",
```
Comment (alvrs) [inline packages/codegen-utils/package.json:2]: nit - should we call this package `common-codegen`? (Also wanna rename the `utils` package to `common`, so it makes more sense for it to also include shared types etc in addition to "utils" in the sense of utility functions)

File: packages/codegen-utils/tsconfig.vitest-temp.json
Comment (alvrs) [inline packages/codegen-utils/tsconfig.vitest-temp.json:1]: ooc why do we need a separate tsconfig for vitest?

File: packages/codegen-utils/vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    sourcemap: true,
+  },
+  /**
+   * @see https://vitest.dev/config/
+   */
+  test: {
+    environment: "node",
+    globals: true,
+  },
+  plugins: [(typescript as any)(), peerDepsExternal()],
```
Comment (alvrs) [inline packages/codegen-utils/vite.config.ts:26]: why do we need to cast to `any` here?

File: pnpm-lock.yaml
Comment (alvrs) [inline pnpm-lock.yaml:1]: weird, @holic @roninjin10 any idea what might cause this?

**Review by alvrs — COMMENTED** [review]
nothing really blocking, but I'd like to wait for @holic or @roninjin10 to have a look at what might cause the `pnpm-lock.yaml` diff

File: packages/codegen-utils/tsconfig.vitest-temp.json
Comment (dk1a) [reply packages/codegen-utils/tsconfig.vitest-temp.json:1]: we don't, removed and gitignored it. That seems to be a weird consequence of using `--passWithNoTests`

File: packages/codegen-utils/vite.config.ts (hunk 2 above)
Comment (dk1a) [reply packages/codegen-utils/vite.config.ts]: I just copied from @roninjin10's PR, the plugin has bad types for some reason

**Review by roninjin10 — COMMENTED** [review]
lockfile just looks liek prettier formatted it

**Review by alvrs — DISMISSED** [review]
Happy to merge as is, but would you mind removing the `pnpm-lock` changes? Adding prettier formatting to .yaml files in https://github.com/latticexyz/mud/pull/629, which should hopefully normalise these changes before committing them

**Comment by holic** [comment]
what pnpm version are you on?

**Review by holic — APPROVED** [review]

**Pull request merged**